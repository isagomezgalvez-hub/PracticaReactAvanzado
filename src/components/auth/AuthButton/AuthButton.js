import { Link } from 'react-router-dom';
import T from 'prop-types';

import { ConfirmationButton } from '../../shared';


//conectar el componente con Redux
import { connect } from 'react-redux';
import { getIsLogged } from '../../../store/selectors';
import { logoutAction } from '../../../actions/authLogin';

const AuthButton = ({ onLogout, isLogged }) => {
  const handleLogoutConfirm = async () => {
    onLogout()
  };

  return isLogged ? (
    <ConfirmationButton
      confirmation="Are you sure?"
      onConfirm={handleLogoutConfirm}
    >
      Logout
    </ConfirmationButton>
  ) : (
    <Link to="/login">Login</Link>
  );
};

AuthButton.propTypes = {
  onLogout: T.func.isRequired,
  isLogged: T.bool,
};

AuthButton.defaultProps = {
  isLogged: false,
};

const mapStateToProps = (state)=>({
  isLogged:getIsLogged(state)
})

const mapDispatchToProps = (dispatch) => ({
  onLogout: () => dispatch(logoutAction()),
})

export default connect(mapStateToProps, mapDispatchToProps)(AuthButton);
