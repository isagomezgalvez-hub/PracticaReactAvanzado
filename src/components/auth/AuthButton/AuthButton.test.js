import { shallow } from 'enzyme';
import { AuthButton } from './AuthButton';

describe('AuthButton ', () => {
	
	const props = {
		isLogged: true,
		onLogout: jest.fn(),
	};

	const render = () => shallow(<AuthButton {...props} />);

	test('should render', () => {
		const wrapper = render()
		expect(wrapper.exists()).toBe(true)
	})
	test('should render snapshot when isLogged', () => {
		const wrapper = render()
		expect(wrapper).toMatchSnapshot();
	});

	test('should render snapshot when is not Logged', () => {
		const wrapper = render()
		wrapper.setProps({ isLogged: false });
		expect(wrapper).toMatchSnapshot();
	});

	test('should call onLogout', () => {
		const wrapper = render()
		wrapper.find('ConfirmationButton').props().onConfirm();
		expect(props.onLogout).toHaveBeenCalled();
	});
});