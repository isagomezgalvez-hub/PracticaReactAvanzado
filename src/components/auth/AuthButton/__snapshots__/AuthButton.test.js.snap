// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton  should render snapshot when is not Logged 1`] = `
<Link
  to="/login"
>
  Login
</Link>
`;

exports[`AuthButton  should render snapshot when isLogged 1`] = `
<ConfirmationButton
  confirmation="Are you sure?"
  onConfirm={[Function]}
>
  Logout
</ConfirmationButton>
`;
